packages/blockchain-extension/extension/commands/importSmartContractPackageCommand.ts

```diff
@@ -49,6 +49,29 @@ export async function importSmartContractPackageCommand(): Promise<void> {
         const packageName: string = path.basename(packagePath);
 
         resolvedPkgDir = path.join(resolvedPkgDir, packageName);
+
+        let existingPackages: string[] = await fs.readdir(packagePath);
+
+        const nameRegex: RegExp = new RegExp(/(.+?)(\.tar\.gz|\.tgz)/gm);
+
+        const result: RegExpExecArray = nameRegex.exec(packageName);
+
+        const newName: string = result[1];
```

Comment:
If the package name isn't matched, then might `result` be of length 0?

Reply:
it will always be matched as you can only import tar.gz or tgz files